File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/roleassignment/fetchoptions/RoleAssignmentFetchOptions.java
/*
 * Copyright 2017 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.roleassignment.fetchoptions;

import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.authorizationgroup.fetchoptions.AuthorizationGroupFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptionsToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.fetchoptions.PersonFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.project.fetchoptions.ProjectFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.roleassignment.RoleAssignment;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.space.fetchoptions.SpaceFetchOptions;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * 
 *
 * @author Franz-Josef Elmer
 */
@JsonObject("as.dto.roleassignment.fetchoptions.RoleAssignmentFetchOptions")
public class RoleAssignmentFetchOptions extends FetchOptions<RoleAssignment> implements Serializable
{
    private static final long serialVersionUID = 1L;
    
    @JsonProperty
    private PersonFetchOptions user;
    
    @JsonProperty
    private AuthorizationGroupFetchOptions authorizationGroup;
    
    @JsonProperty
    private SpaceFetchOptions space;

    @JsonProperty
    private ProjectFetchOptions project;
    
    @JsonProperty
    private RoleAssignmentSortOptions sort;
    
    public PersonFetchOptions withUser()
    {
        if (user == null)
        {
            user = new PersonFetchOptions();
        }
        return user;
    }
    
    public PersonFetchOptions withUserUsing(PersonFetchOptions fetchOptions)
    {
        return user = fetchOptions;
    }
    
    public boolean hasUser()
    {
        return user != null;
    }
    
    public AuthorizationGroupFetchOptions withAuthorizationGroup()
    {
        if (authorizationGroup == null)
        {
            authorizationGroup = new AuthorizationGroupFetchOptions();
        }
        return authorizationGroup;
    }
    
    public AuthorizationGroupFetchOptions withAuthorizationGroupUsing(AuthorizationGroupFetchOptions fetchOptions)
    {
        return authorizationGroup = fetchOptions;
    }
    
    public boolean hasAuthorizationGroup()
    {
        return authorizationGroup != null;
    }

    public SpaceFetchOptions withSpace()
    {
        if (space == null)
        {
            space = new SpaceFetchOptions();
        }
        return space;
    }

    public SpaceFetchOptions withSpaceUsing(SpaceFetchOptions fetchOptions)
    {
        return space = fetchOptions;
    }

    public boolean hasSpace()
    {
        return space != null;
    }

    public ProjectFetchOptions withProject()
    {
        if (project == null)
        {
            project = new ProjectFetchOptions();
        }
        return project;
    }
    
    public ProjectFetchOptions withProjectUsing(ProjectFetchOptions fetchOptions)
    {
        return project = fetchOptions;
    }
    
    public boolean hasProject()
    {
        return project != null;
    }
    
    @Override
    public RoleAssignmentSortOptions sortBy()
    {
        if (sort == null)
        {
            sort = new RoleAssignmentSortOptions();
        }
        return sort;
    }

    @Override
    public RoleAssignmentSortOptions getSortBy()
    {
        return sort;
    }

    @Override
    protected FetchOptionsToStringBuilder getFetchOptionsStringBuilder()
    {
        FetchOptionsToStringBuilder f = new FetchOptionsToStringBuilder("RoleAssignment", this);
        return f;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/roleassignment/fetchoptions/RoleAssignmentFetchOptions.js
define([ "require", "stjs", "as/dto/common/fetchoptions/FetchOptions", "as/dto/person/fetchoptions/PersonFetchOptions", 
         "as/dto/authorizationgroup/fetchoptions/AuthorizationGroupFetchOptions", "as/dto/space/fetchoptions/SpaceFetchOptions", 
         "as/dto/project/fetchoptions/ProjectFetchOptions", "as/dto/roleassignment/fetchoptions/RoleAssignmentSortOptions" ], function(require, stjs,
		FetchOptions) {
	var RoleAssignmentFetchOptions = function() {
	};
	stjs.extend(RoleAssignmentFetchOptions, FetchOptions, [ FetchOptions ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.roleassignment.fetchoptions.RoleAssignmentFetchOptions';
		constructor.serialVersionUID = 1;
		prototype.user = null;
		prototype.authorizationGroup = null;
		prototype.space = null;
		prototype.sort = null;
		prototype.project = null;
		prototype.withUser = function() {
			if (this.user == null) {
				var PersonFetchOptions = require("as/dto/person/fetchoptions/PersonFetchOptions");
				this.user = new PersonFetchOptions();
			}
			return this.user;
		}
		prototype.withUserUsing = function(fetchOptions) {
			return this.user = fetchOptions;
		}
		prototype.hasUser = function() {
			return this.user != null;
		}
		prototype.withAuthorizationGroup = function() {
			if (this.authorizationGroup == null) {
				var AuthorizationGroupFetchOptions = require("as/dto/authorizationgroup/fetchoptions/AuthorizationGroupFetchOptions");
				this.authorizationGroup = new AuthorizationGroupFetchOptions();
			}
			return this.authorizationGroup;
		}
		prototype.withAuthorizationGroupUsing = function(fetchOptions) {
			return this.authorizationGroup = fetchOptions;
		}
		prototype.hasAuthorizationGroup = function() {
			return this.authorizationGroup != null;
		}
		prototype.withSpace = function() {
			if (this.space == null) {
				var SpaceFetchOptions = require("as/dto/space/fetchoptions/SpaceFetchOptions");
				this.space = new SpaceFetchOptions();
			}
			return this.space;
		};
		prototype.withSpaceUsing = function(fetchOptions) {
			return this.space = fetchOptions;
		};
		prototype.hasSpace = function() {
			return this.space != null;
		};
		prototype.withProject = function() {
			if (this.project == null) {
				var ProjectFetchOptions = require("as/dto/project/fetchoptions/ProjectFetchOptions");
				this.project = new ProjectFetchOptions();
			}
			return this.project;
		};
		prototype.withProjectUsing = function(fetchOptions) {
			return this.project = fetchOptions;
		};
		prototype.hasProject = function() {
			return this.project != null;
		};
		prototype.sortBy = function() {
			if (this.sort == null) {
				var RoleAssignmentSortOptions = require("as/dto/roleassignment/fetchoptions/RoleAssignmentSortOptions");
				this.sort = new RoleAssignmentSortOptions();
			}
			return this.sort;
		};
		prototype.getSortBy = function() {
			return this.sort;
		};
	}, {
		user : "PersonFetchOptions",
		authorizationGroup : "AuthorizationGroupFetchOptions",
		space : "SpaceFetchOptions",
		project : "ProjectFetchOptions",
		sort : "RoleAssignmentSortOptions"
	});
	return RoleAssignmentFetchOptions;
})